### ISSUE TITLE
Project is't compiling if Display name was changed

### ISSUE THREAD

**Issue by albertchubakov** [mac-cain13/R.swift#581]
Project is't compiling if Display name was changed. Xcode 11.1

**Comment by mac-cain13** [comment]
Thanks for using R.swift.

How is this error related to R.swift? What is the error you're getting? What did you try to resolve it? Is the project compiling if you disable R.swift?

**Comment by albertchubakov** [comment]
> Thanks for using R.swift.
> 
> How is this error related to R.swift? What is the error you're getting? What did you try to resolve it? Is the project compiling if you disable R.swift?

Hi. When i changed "Display name" parameter in Xcode, then select clean build folder, then build project with new Display name - i got error "No such module TargetName" in import line in R.generated.swift file

**Comment by yansaid** [comment]
> > Thanks for using R.swift.
> > How is this error related to R.swift? What is the error you're getting? What did you try to resolve it? Is the project compiling if you disable R.swift?
> 
> Hi. When i changed "Display name" parameter in Xcode, then select clean build folder, then build project with new Display name - i got error "No such module TargetName" in import line in R.generated.swift file

I have the same issue

**Comment by mac-cain13** [comment]
Are you sure you _only_ changed the display name? Changing the display name *should not* change the name of the module we need to import.

Also normally there should not be an import for your own module in the generated file.

I could not reproduce this with my projects. Could you provide an example?

**Comment by yansaid** [comment]
I remove all to solve the solution,

**Comment by tomlokhorst** [comment]
R.swift will generate a `import Something` for each module named in a xib Custom Class.

If that module doesn't exist, the generated code won't compile. However, without R.swift the specified custom class also won't be loaded (because the module doesn't exist).

**Comment by tomlokhorst** [comment]
Closing due to inactivity.

If you do have further comments, feel free to leave them here.

**Comment by akatao** [comment]
I'm having the same error, when I changed the App Display Name.
I tried to disable R.swift when compiling but the app show errors on all places that Im using the R.swift.

**Comment by akatao** [comment]
> Thanks for using R.swift.
> 
> How is this error related to R.swift? What is the error you're getting? What did you try to resolve it? Is the project compiling if you disable R.swift?

I want to know why the project won't compile when I changed the App Display name.
Because the target name stays the same.

**Comment by akatao** [comment]
> Are you sure you _only_ changed the display name? Changing the display name _should not_ change the name of the module we need to import.
> 
> Also normally there should not be an import for your own module in the generated file.
> 
> I could not reproduce this with my projects. Could you provide an example?

Im sure that I only changed the Display name.
I changed the display name and the target name no changed.
Normally there should not be an import for my own module in the generated file, but why my module is imported in the R.generated?
[
<img width="984" alt="Screen Shot 2020-06-02 at 21 48 27" src="https://user-images.githubusercontent.com/3901891/83583460-ce792500-a51a-11ea-9e3b-e955d19ec94f.png">
<img width="845" alt="Screen Shot 2020-06-02 at 21 44 53" src="https://user-images.githubusercontent.com/3901891/83583494-e355b880-a51a-11ea-860b-0f7b56748bcd.png">

](url)
